
version: '3'
services:
  network-monitoring-db:
    image: postgres:11.1
    shm_size: 4gb
    environment:
      POSTGRES_USER: postgres
      POSTGRES_DB: audius_network_monitoring
    volumes:
      - postgres_data_network_monitoring_2:/var/lib/postgresql/data/
    ports:
      - '${audius_nm_db_port}:5432'
    networks:
      - audius_dev
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  network-monitoring-push-gateway:
    image: prom/pushgateway
    ports:
      - 9091:9091
    networks:
      - audius_dev
  
  grafana:
    build:
      context: grafana
      args:
        - GRAFANA_VERSION=latest
        - GF_INSTALL_IMAGE_RENDERER_PLUGIN=true
    user: 0:0
    env_file:
      - grafana/.env
    links:
      - prometheus:prometheus
    ports: 
      - "3000:3000"
    restart: always
    volumes:
      - ./data/grafana:/var/lib/grafana
    networks:
      - audius_dev

  prometheus:
    image: prom/prometheus:v2.33.4
    user: 0:0
    extra_hosts:
      - "host.docker.internal:host-gateway"
    ports:
      - "9090:9090"
    volumes:
      - ./data/prometheus:/prometheus
      - ./prometheus/config/prometheus.yml:/etc/prometheus/prometheus.yml
    networks:
      - audius_dev

  network-monitoring:
    build: '.'
    depends_on:
      - network-monitoring-db
      - network-monitoring-push-gateway
    networks:
      - audius_dev

volumes:
  postgres_data_network_monitoring_2:

networks:
  audius_dev:
    external: true
